types/react-swipeable/index.d.ts

```diff
@@ -1,51 +1,39 @@
-    interface onSwipingCallback {
-        (event: React.TouchEvent<any>, deltaX: number, deltaY: number, absX: number, absY: number, velocity: number): void;
-    }
-
-    interface OnSwipedCallback {
-        (event: React.TouchEvent<any>, deltaX: number, deltaY: number, isFlick: boolean, velocity: number): void;
-    }
-
-    interface OnSwipedDirectionCallback {
-        (event: React.TouchEvent<any>, delta: number, isFlick: boolean): void;
-    }
-
-    interface OnSwipingDirectionCallback {
-        (event: React.TouchEvent<any>, delta: number): void;
-    }
-
-    interface Props {
-        onSwiped?: OnSwipedCallback;
-        onSwiping?: onSwipingCallback;
-        onSwipingUp?: OnSwipingDirectionCallback;
-        onSwipingRight?: OnSwipingDirectionCallback;
-        onSwipingDown?: OnSwipingDirectionCallback;
-        onSwipingLeft?: OnSwipingDirectionCallback;
-        onSwipedUp?: OnSwipedDirectionCallback;
-        onSwipedRight?: OnSwipedDirectionCallback;
-        onSwipedDown?: OnSwipedDirectionCallback;
-        onSwipedLeft?: OnSwipedDirectionCallback;
-        flickThreshold?: number;
-        delta?: number;
-        preventDefaultTouchmoveEvent?: boolean;
-        nodeName?: string;
-    }
-
-    interface ReactSwipeable extends React.ComponentClass<Props> { }
-
+// TypeScript Version: 2.2
+
+import * as React from 'react';
+
+type onSwipingCallback = (event: React.TouchEvent<HTMLElement>, deltaX: number, deltaY: number, absX: number, absY: number, velocity: number) => void;
+type OnSwipedCallback = (event: React.TouchEvent<HTMLElement>, deltaX: number, deltaY: number, isFlick: boolean, velocity: number) => void;
+type OnSwipedDirectionCallback = (event: React.TouchEvent<HTMLElement>, delta: number, isFlick: boolean) => void;
+type OnSwipingDirectionCallback = (event: React.TouchEvent<HTMLElement>, delta: number) => void;
+type OnTapCallback = (event: React.TouchEvent<HTMLElement>) => void;
+
+interface Props extends React.ClassAttributes<ReactSwipeable>, React.HTMLAttributes<HTMLElement> {
+    onSwiped?: OnSwipedCallback;
+    onSwiping?: onSwipingCallback;
+    onSwipingUp?: OnSwipingDirectionCallback;
+    onSwipingRight?: OnSwipingDirectionCallback;
+    onSwipingDown?: OnSwipingDirectionCallback;
+    onSwipingLeft?: OnSwipingDirectionCallback;
+    onSwipedUp?: OnSwipedDirectionCallback;
+    onSwipedRight?: OnSwipedDirectionCallback;
+    onSwipedDown?: OnSwipedDirectionCallback;
+    onSwipedLeft?: OnSwipedDirectionCallback;
+    onTap?: OnTapCallback;
+    flickThreshold?: number;
+    delta?: number;
+    preventDefaultTouchmoveEvent?: boolean;
+    stopPropagation?: boolean;
+    nodeName?: string;
+    trackMouse?: boolean;
+    children?: React.ReactNode;
 }
 
+declare class ReactSwipeable extends React.Component<Props, void> {}
+declare namespace ReactSwipeable {}
 
-declare module "react-swipeable" {
-    let module: ReactSwipeableModule.ReactSwipeable;
-    export = module;
-}
+export = ReactSwipeable;
```

Comment:
Because you've unwrapped all of the types from the namespace, the defined types can no longer be externally referenced.